src/boot/efi/console.c

```diff
@@ -266,7 +266,8 @@ EFI_STATUS console_set_mode(int64_t mode) {
 
                 mode = MAX(CONSOLE_MODE_RANGE_MIN, ST->ConOut->Mode->Mode);
                 do {
-                        mode = (mode + 1) % ST->ConOut->Mode->MaxMode;
+                        if (++mode >= ST->ConOut->Mode->MaxMode)
+                                mode = 0;
```

Comment:
i am sorry, but why? this is longer and less readable i would say. It stuffs a side-effect into a comparison, with is hardly an improvement.

I am not sure why we would make such a change, it doesn't look like any improvement?

Reply:
hmm, unanswered, but marked as resolved?

Reply:
modulo can be expensive on RISC architectures where the dividend is not a constant that is a power of 2.

Reply:
this is not a performance senstive codepaths, not at all